sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java

```diff
@@ -722,6 +727,21 @@ SearchClientBuilder getSearchClientBuilder(String indexName) {
         return withContext(context -> deleteSynonymMapWithResponse(synonymMap.getName(), etag, context));
     }
 
+    /**
+     * Helper method to build list of {@link SearchField}.
+     *
+     * @param model The model class where {@link SearchField} converts from.
+     * @param options The option property bag.
+     * @return The list {@link SearchField} for search index schema.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public static Mono<List<SearchField>> buildSearchField(Class<?> model, FieldBuilderOptions options) {
```

Comment:
Not a huge fan of this being an asynchronous return and we have prior art which has non-service APIs returning synchronously in an asynchronous client. `getSearchAsyncClient` on this client is synchronous, I think this should be made synchronous as well.

@srnagar @JonathanGiles, let me know if you disagree.

Reply:
I can have synchronous call. Is it necessary to have the function in both sync and async client?

Reply:
For convenience I would say yes.